### ISSUE TITLE
Dragon Breath leaves empty Glass Bottle in Brewing Stand when used

### ISSUE THREAD

**Issue by DDaMAGEr** [FabricMC/fabric#2873]
Dragon Breath leaves empty Glass Bottle in ingredient slot of Brewing Stand when used.

For some strange reason, this previously happened with Forge pretty long ago, and now it happens with Fabric, but only when Fabric API is present among mods. This is pretty bad, because it may break brewing automatization, as there seems to be no way of removing empty glass bottle from ingredient slot automatically. Tested it on 1.19.3, fabric version 0.14.13, fabric api 0.73.0+1.19.3 (seems to be the latest release), the problem exists.

**Comment by haykam821** [comment]
I can reproduce the issue by running the following command and observing the item in the upper slot of the brewing stand:

```mcfunction
setblock ~2 ~ ~ minecraft:brewing_stand{Fuel: 20s, Items: [{Slot: 1, id: "minecraft:splash_potion", Count: 1}, {Slot: 3, id: "minecraft:dragon_breath", Count: 1}]}
```

Initial | Expected | Actual
--- | --- | ---
<img width="579" alt="image" src="https://user-images.githubusercontent.com/24855774/215634271-431ef831-4ff9-41ff-bdf9-48e36aa96db8.png"> | <img width="577" alt="image" src="https://user-images.githubusercontent.com/24855774/215634337-b2eb6e9f-4368-4834-b75a-020fecdfdcbc.png"> | <img width="579" alt="image" src="https://user-images.githubusercontent.com/24855774/215634308-25aa92b3-081b-4702-8769-b1b6bfaf6148.png">

**Comment by Shnupbups** [comment]
Funnily enough, this seems to be actually fixing a vanilla bug. But I'd agree that we should un-fix said bug to retain vanilla parity.

**Comment by modmuss50** [comment]
> For some strange reason, this previously happened with Forge pretty long ago

I have just tested, this is still an issue in Forge. I have been struggling to fix this so vanilla behaviour is unchanged but this works with modded items, had hoped I could get an idea of how to fix it from looking at Forge but nope!

> This is pretty bad, because it may break brewing automatization

Yes this is very bad, we should never fix vanilla bugs. I have prepared a fix here: https://github.com/FabricMC/fabric/pull/2874 this does apply the same vanilla bug to our modded items.

The vanilla bug can clearly be reproduced by putting 2 dragons breath in the top slot, and a single potion in the bottom slot. On the first brew an empty bottle is spawned into the world, on the 2nd the empty bottle is lost.
